### ISSUE TITLE
Implement valuable-serde crate

### ISSUE THREAD

**Issue by carllerche** [tokio-rs/valuable#17]
It should be possible to implement Serde's `Serialize` trait for `valuable::Value`. This should enable using Serde's existing serializer ecosystem with valuable.

**Comment by carllerche** [comment]
@taiki-e is this something you would be interested in taking on? If it is, you can assign the issue to yourself.

**Comment by taiki-e** [comment]
I've tried it before: https://github.com/tokio-rs/valuable/blob/84c6992b55981181d7b75bb79fcf26428994a019/valuable/examples/serde.rs

If I remember correctly:
- As for implementing Seriarize on Value: I changed some fields such as StructDef::name from `&'a str` to `&'static str`, but there were no other major problems.
- As for implementing Valuable on impl Seriarize: It was very difficult and I could not get it to work well.

I will try to update it to the current valuable trait and see if it works.
